### DISCUSSION TITLE
are there any functional example with Ionic 7 and Angular?

### DISCUSSION THREAD

**Discussion by ortaprogram** [capacitor-community/barcode-scanner#255] [Q&A]
So complicated to generate a functional app with this workaround :c 
Im just trying to generate a QR reader and show the result. As simple as that. But i can't see the camera opening and the documentation is not good enough. Could anyone share a DEMO rep or example working please? 

ionic 7.1.1
Capacitor 5
Angular

**Comment by thegnuu** [comment]
We will not add any demo projects to this repository, there are way too many possible configurations and mixes of frameworks that would allow us to maintain demo projects. This would generate way too much work.

The documentation will be updated for the upcoming release of v5 since the interfaces will change anyway and the setup should be more straightforward due to generalized functions.

The only thing we cannot change is the handling of the DOM, any dev needs to verify his own setup and make changes accordingly, due to the nature of the plugin (on iOS and Android the camera view will always be rendered behind the web view).

**Comment by amav96** [comment]
try to find <video id="video"/> on your DOM and set z-index:9999, i had te same problem

`
          const startScan = async () => {
        
          // Check camera permission
          // This is just a simple example, check out the better checks below
          await BarcodeScanner.checkPermission({ force: true });
        
          // make background of WebView transparent
          // note: if you are using ionic this might not be enough, check below
          BarcodeScanner.hideBackground();
        
           // Configura el intervalo para verificar la presencia del elemento 'video'
           let intervalId = setInterval(() => {
            const videoElement = document.getElementById('video');
            if (videoElement) {
              const videoParent = videoElement.parentNode;
              if (videoParent && videoParent.style.zIndex !== '9999') {
                // Aplica estilos al padre del video si aún no se ha aplicado
                videoParent.style.zIndex = '9999';
                // Otros estilos que desees aplicar al div padre
        
                // Detén el intervalo después de aplicar los estilos
                clearInterval(intervalId);
              }
            }
          }, 800); // Intervalo de 2 segundos (ajusta según tus necesidades)

          const result : any = await BarcodeScanner.startScan(); // start scanning and wait for a result
        
         
        
          // if the result has content
          if (result.hasContent) {
            const { id } = JSON.parse(result.content)
            if(id){
                itemForm.value.track_id = id
            }
          }
        
          stopScan()
};

`
